design-system/packages/fields/src/utils/dateFormatters.ts

```diff
@@ -0,0 +1,37 @@
+import { format, formatISO } from 'date-fns';
+import { DateType } from '../../types';
+
+/**
+ * Un-formatted date for server side storage (ISO8601), like '2019-09-18'
+ */
+export const formatDateType = (date: Date): DateType => {
+  return formatISO(date, { representation:"date" });
```

Comment:
nice QoL from @mitchellhamilton, DatePicker should return just the date representation and not the whole ISO string. Format function changed to reflect this.